NeuralNet.js

```diff
@@ -0,0 +1,171 @@
+class NeuralNet {
+    constructor(options) {
+        this.config = {
+            inputs: options.inputs || 3,
+            outputs: options.outputs || 2,
+            debug: options.debug || true,
+            outputLabels: options.outputLabels || [],
+            inputLabels: options.inputLabels || [],
+            batchSize:options.batchSize || 64,
+            epochs: options.epochs || 32,
+            NO_VAL: null
+        }
+        this.config.NO_VAL = this.config.outputs;
+        this.model = null;
+    }
+
+
+    async createModel() {
+
+        this.model = tf.sequential();
+
+        const hidden = tf.layers.dense({
+            units: 16,
+            inputShape: [this.config.inputs],
+            activation: 'relu'
+        });
+        
+        const output = tf.layers.dense({
+            units: this.config.outputs,
+            activation: 'softmax'
+        });
+        this.model.add(hidden);
+        // this.model.add(tf.layers.dense({units: 10, activation: 'relu'}));
+        this.model.add(output);
+
+        this.model.compile({
+            optimizer: tf.train.adam(),
+            // This loss breaks things?
+            // loss: 'sparseCategoricalCrossentropy',
+            loss: 'categoricalCrossentropy',
+            metrics: ['accuracy'],
+        });
+
+        if (this.config.debug) {
+            tfvis.show.modelSummary({
+                name: 'Model Summary'
+            }, this.model);
+        }
+    }
+
+    async loadData(DATA_URL, dataType) {
+        const outputLabel = this.config.outputLabels[0];
```

Comment:
How we handle the user specifying what is a "label" in the data for both inputs and outputs is tricky.

Reply:
Totally. We'll have to put some more thoughts on this. Maybe we create a hashmap that stores the key:values ?